src/main/kotlin/graphics/scenery/DFTParser.kt

```diff
@@ -0,0 +1,132 @@
+                    gridDimensions[counter-3] =  (line.trim().split("\\s+".toRegex())[0]).toInt()
+                    gridSpacings[counter-3] = (line.trim().split("\\s+".toRegex())[counter-2]).toFloat()
+                }
+                else->  {
+                    if (counter == 6)
+                    {
+                        unitCellDimensions[0] = (gridSpacings[0]*gridDimensions[0])+unitCellOrigin[0]
+                        unitCellDimensions[1] = (gridSpacings[1]*gridDimensions[1])+unitCellOrigin[1]
+                        unitCellDimensions[2] = (gridSpacings[2]*gridDimensions[2])+unitCellOrigin[2]
+                        electronicDensity = Array(gridDimensions[0], { Array(gridDimensions[1],
+                            { FloatArray(gridDimensions[2]) } ) } )
+                    }
+                    // Parsing atomic positions.
+                    if (counter < 6+numberOfAtoms){
+                        atomicPositions[counter-6] = Vector3f((line.trim().split("\\s+".toRegex())[2]).toFloat(),
+                                                              (line.trim().split("\\s+".toRegex())[3]).toFloat(),
+                                                              (line.trim().split("\\s+".toRegex())[4]).toFloat())
+                    }
+
+                    // Parsing volumetric data.
+                    // A possible optimization here would be to read this into a 1D array. We cannot directly
+                    // read it into the byte buffer, because we don't know max/min values a-priori.
+                    if (counter >= 6+numberOfAtoms) {
+                        for (value in (line.trim().split("\\s+".toRegex()))) {
+                            // Cube files should be in Fortran (z-fastest ordering).
+                            electronicDensity[xcounter][ycounter][zcounter] = (value).toFloat()
+                            if (electronicDensity[xcounter][ycounter][zcounter] > maxDensity) {
+                                maxDensity = electronicDensity[xcounter][ycounter][zcounter]
+                            }
+                            if (electronicDensity[xcounter][ycounter][zcounter] < minDensity) {
+                                minDensity = electronicDensity[xcounter][ycounter][zcounter]
+                            }
+
+                            zcounter++
+                            if (zcounter == gridDimensions[2]) {
+                                zcounter = 0
+                                ycounter++
+                                if (ycounter == gridDimensions[1]) {
+                                    ycounter = 0
+                                    xcounter++
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+            counter++
+        }
+        // Converting to byte buffer.
+        counter = 0
+        electronicDensityUInt =  MemoryUtil.memAlloc((gridDimensions[0]*
```

Comment:
`memAlloc` creates off-JVM-heap memory, so that memory at some point needs to be released again. Best if you class implements the `AutoCloseable` interface, and you `memFree` the memory in the `close()` routine 👍

Reply:
Oh yes, good point, I totally forgot to do that... I just implemented it.